src/pages/Help.vue

```diff
@@ -194,11 +141,26 @@ export default {
           .map(([key, value]) => key)
           .join(',')
 
+        // Fetch the Adress of the user
+        if (this.query.address.city === '') {
```

Comment:
This is the wrong place to check for the address. All the code related to prefilling the address in the query should go into mounted, you shouldn't have to touch categoryIds and distance and the address provided via route params should take precedence over the address coming from the user profile.

`this.query.address.city === ''` will always be false, because `this.query.address` is a string not an object by the way.

We already have the address in the store:
```
this.auth.streetNumber
this.auth.street
this.auth.zip
this.auth.city
```

You should check if those are falsy.
If they are falsy, then show the banner.

Please also rename the `showBanner` variable to `isUserAddressMissing` and make the default false.

Reply:
You're right `this.query.address` is always false.
However,
```
this.auth.streetNumber
this.auth.street
this.auth.zip
this.auth.city
```
This is not in the store, or am I missing something?

Reply:
Our store handling is a bit funky, but it should be.

```
  computed: {
    auth: {
      get() {
        return Object.assign({}, this.$store.state.auth.data)
      },
      set(val) {
        this.$store.commit('auth/updateData', val)
      }
    }
  },
```

Reply:
This is my console.log
```
{token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1Z…zMTh9.3pdn5e8Vfj8QYh_IOnmmWW1HmgKaq6Ye3Zf5RMaUp2c", firstname: "bla", lastname: "bla", email: "bla@bla.bla", phoneNumber: "", …}
authenticated: true
createdAt: ""
email: "bla@bla.bla"
firstname: "bla"
id: "5eadbcb8e58af22bc4b57f8d"
lastname: "bla"
phoneNumber: ""
picture: ""
token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ZWFkYmNiOGU1OGFmMjJiYzRiNTdmOGQiLCJlbWFpbCI6ImJsYUBibGEuYmxhIiwiaWF0IjoxNTg4NDk2OTE4LCJleHAiOjE1ODg1ODMzMTh9.3pdn5e8Vfj8QYh_IOnmmWW1HmgKaq6Ye3Zf5RMaUp2c"
updatedAt: ""
verified: true
__proto__: Object
```

In the `src/store/auth/state.js` the address is not in there.

Reply:
You're right. 

1. Please add address: { number: undefined, street: undefined:, .... } to the state and set the auth state in Login.vue.

2. Remove the fetchUserData logic from the profile, because it should already be available in the store at that point. 

3. Check in Register.vue if the response from authApi already includes all the data, If so then there is no need to callApi('users/me') either.

4. In ProfileButton.vue in the logout function, also unset the address in the auth store.

5. In Verify.vue, the verifyEmail function can also only be called if the user is login, you should also refactor callApi('users/me) away.

Reply:
Point 3 is not possible because there is no field for registering the Address in the Registration process